### ISSUE TITLE
Error on AddressEntry.get_Type() 

### ISSUE THREAD

**Issue by phw198** [phw198/OutlookGoogleCalendarSync#57]
Hi there,  
  
first of all, thanks for your sync application and all the work you put into
it. I have been using OGCS for quite a while now, synching worked like a charm
for quite a while but since some month I get recurring sync errors for
apppointments that contain emails that are not part of our email domain (at
least it seems so). Here's an example from the OGcalsync.log file showing the
problem:  
  
2017-02-20 21:12:44,015 FINE 5 OutlookGoogleCalendarSync.OutlookNew [0] -
Determining email of recipient: Someone Else [
(someone@else.com)&lt;someone@else.com&gt; 2017-02-20
](mailto:\(someone@else.com\)<someone@else.com>

2017-02-20) 21:12:44,218 INFO 5 OutlookGoogleCalendarSync.MainForm [0] -
WARNING: Appointment update failed.  
Fehler beim Ausführen der Operation.  
2017-02-20 21:12:44,218 ERROR 5 OutlookGoogleCalendarSync.OutlookCalendar [0]
- bei Microsoft.Office.Interop.Outlook.AddressEntry.get_Type()  
bei OutlookGoogleCalendarSync.OutlookNew.GetRecipientEmail(Recipient
recipient)  
bei
OutlookGoogleCalendarSync.OutlookCalendar.UpdateCalendarEntry(AppointmentItem&amp;
ai, Event ev, Int32&amp; itemModified, Boolean forceCompare)  
bei
OutlookGoogleCalendarSync.OutlookCalendar.UpdateCalendarEntries(Dictionary`2
entriesToBeCompared, Int32&amp; entriesUpdated)  
  
I got 5 errors today (and have to confirm every one of them), each of them
starting with the "Determing email ..." line and foreign email addresses and
it's quite annoying if you have to do this ever hour. I am using the latest
version 2.3.3.0, the error still occurs in it.  
Will donate money for some beers if you find a fix for the problem :-)  
  
Best, Sascha  

Work Item Details
--------------------
**Original CodePlex Issue:**	[Issue 462](http://outlookgooglecalendarsync.codeplex.com/workitem/462)
**Status:**	Active
**Reason Closed:**	Unassigned
**Assigned to:**	phw198
**Reported on:**	Feb 20 at 7:29 PM
**Reported by:**	AlexWayhill
**Updated on:**	Mar 5 at 5:45 PM
**Updated by:**	phw198

#### Binary Attachments

[OutlookGoogleCalendarSync.exe](http://outlookgooglecalendarsync.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=outlookgooglecalendarsync&WorkItemId=462&FileAttachmentId=1641115)

**Comment by phw198** [comment]
On *2017-02-26 03:57:52 UTC*, **phw198** commented:

Hi Sascha,

Could you dial up the logging level to "FINE", recreate the error and then provide your log file. Probably don't need the whole thing, just the 50 lines or so before the error happens.

Cheers!

**Comment by phw198** [comment]
On *2017-02-26 23:06:53 UTC*, **AlexWayhill** commented:

Hi and thanks for your reply! The log level was already set to "All", I now changed it to "Ultra-Fine". The logged entry is still the same as above:

2017-02-20 21:12:54,847 FINE 5 OutlookGoogleCalendarSync.OutlookNew [0] - Determining email of recipient: Bxxx Kxxx
(kxxx@company.de)<kxxx@company.de> 2017-02-20 21:12:55,064 INFO 5 OutlookGoogleCalendarSync.MainForm [0] - WARNING: Appointment update failed.
Fehler beim Ausführen der Operation.
2017-02-20 21:12:55,064 ERROR 5 OutlookGoogleCalendarSync.OutlookCalendar [0] - bei Microsoft.Office.Interop.Outlook.AddressEntry.get_Type()
bei OutlookGoogleCalendarSync.OutlookNew.GetRecipientEmail(Recipient recipient)
bei OutlookGoogleCalendarSync.OutlookCalendar.UpdateCalendarEntry(AppointmentItem& ai, Event ev, Int32& itemModified, Boolean forceCompare)
bei OutlookGoogleCalendarSync.OutlookCalendar.UpdateCalendarEntries(Dictionary`2 entriesToBeCompared, Int32& entriesUpdated)

Because of security concerns I can't post the full log, but I could mail the log file to you if required.

Thanks and have a nice sunday!

Best, Sascha

**Comment by phw198** [comment]
On *2017-02-28 01:37:28 UTC*, **AlexWayhill** commented:

Update with your latest alpha release 2.3.4:

I unchecked the "Attendees" flag and that solved the problem in my case. It's not fixing the underlying problem but I don't need the attendees anyway on my smartphone/Outlook, I just need the reminders and appointments sync with my private stuff.

I also had a look at the OutlookNew class and at least it seems as if the error should have got caught but the alpha (which seems to be what I just forked) still gives the errors. I am really thankful for the work that was put into OutlookGoogleCalendarSync
 but the code could need an overhaul as there are so many wrapped try-catch/try-finally blocks that it's hard to understand what really causes the problems. If you need some help with restructuring and to create testable code, let me know.

Best, Sascha

**Comment by phw198** [comment]
On *2017-02-28 21:24:44 UTC*, **phw198** commented:

As far as I can tell, it's erroring on
log.Fine("AddressEntry Type: " + addressEntry.Type);
which is kind of crazy as addressEntry.Type should be a string and I've already tested addressEntry isn't null. Anyways, please give attached hotfix v2.3.4.3 a go.


And I agree, the code could do with a bit of an overhaul, but I'm wanting to get out the next Beta release first. To let you into a secret, I'm not a developer by trade and I've pretty much learnt C# from the ground up on this project, so I've no problem with
 any pointers you may have on code restructure :-)

**Comment by phw198** [comment]
On *2017-03-01 04:20:45 UTC*, **AlexWayhill** commented:

This seems to fix it (although I haven't fully checked it as the one attendee that always caused the crash is currently not in any appointment)!

That's a really good job you did there and learning a programming language from scratch (along with gaining the knowledge about all kinds of different mail server formats and interface descriptions) is not the easiest task. I learned quite a few programming
 languages and C# is the newest that I got into but I can do some reorganisation of code so that it becomes more testable, understandable and in that way extensible. I'll wait for the next beta release and could then commit a rewrite, just see for yourself
 if that is any help for you :-)

Thanks for your troubles solving this problem! And as promised some pounds for a beer or two is on its way :-)
